tests/ion_hash_tests.rs

```diff
@@ -224,7 +225,7 @@ fn expected_hash(struct_: &Struct) -> IonResult<Vec<u8>> {
     let identity = if let Some(identity) = struct_.get("identity") {
         identity.as_sequence().expect("`identity` should be a sexp")
     } else {
-        illegal_operation("only identity tests are implemented")?
+        todo!("only identity tests are implemented")
```

Comment:
🗺️ ion-hash-tests was constructing its own `IonResult`s to describe failure modes that didn't come from Ion itself. The error constructor methods are now limited to crate-visibility. Clients like ion-hash-tests should use their own error type.

In ion-hash-tests' case, it was bubbling up the error result and panicking later on. I've changed those failures to an immediate panic. The end result is the same, and it removes the dependency on now-private functionality.